Extension/src/main.ts

```diff
@@ -48,6 +48,24 @@ export async function activate(context: vscode.ExtensionContext): Promise<CppToo
     Telemetry.activate();
     util.setProgress(0);
 
+    // check if the correct offline/insiders vsix is installed on the correct platform
+    let installedPlatform: string = await util.getInstalledBinaryPlatform();
+    if (process.platform !== installedPlatform) {
+        let helpURL: string = "https://github.com/Microsoft/vscode-cpptools/releases";
+        errMsg = localize("native.binaries.not.supported", "The installed VSIX is incompatible with {0}. Download the {0} VSIX from: {1}.", process.platform, helpURL);
```

Comment:
I would like to avoid using "VSIX" in the message since most non-MSFT folks don't refer to the extension that way.  I think a simple message that says something like: "The {0} version of the extension appears to be installed. Please download and install the {1} version of the extension." would be better.  Putting the URL in the message is unnecessary because the button will take you there.

Reply:
Or "The {0} version of the extension is incompatible with your OS"